### PR TITLE
[frameworks] Add `settings` to frameworks

### PR DESCRIPTION
**Pull request by AndyBitz**
Adds a new `settings` property that can be used to retrieve defaults for the framework.

[PRODUCT-944]

[PRODUCT-944]: https://zeit.atlassian.net/browse/PRODUCT-944

### CODE REVIEW

File: packages/frameworks/frameworks.json [hunk 1]
```diff
@@ -174,6 +262,14 @@
           "matchContent": "\"(dev)?(d|D)ependencies\":\\s*{[^}]*\"sirv-cli\":\\s*\".+?\"[^}]*}"
         }
       ]
+    },
+    "settings": {
+      "buildCommand": {
+        "placeholder": "`polymer build` or `build` from `package.json`"
```
Comment (leo) [inline packages/frameworks/frameworks.json]: Not sure if Polymer again here makes sense? It's already being defined above.

**Review by leo — CHANGES_REQUESTED** [review]

File: packages/frameworks/frameworks.json (hunk 1 above)
Comment (AndyBitz) [reply packages/frameworks/frameworks.json]: Copy-Paste mistake, fixed it

**Review by leo — APPROVED** [review]
Shippable!

**Comment by lucleray** [comment]
What about `outputDirectory`?

File: packages/frameworks/frameworks.json [hunk 2]
```diff
@@ -12,6 +12,17 @@
...
+    },
+    "settings": {
+      "buildCommand": {
+        "placeholder": "`next build` or `build` from `package.json`"
+      },
+      "devCommand": {
+        "value": "next dev --port $PORT"
+      },
+      "outputDirectory": {
+        "placeholder": "value from `next.config.js`"
```
Comment (leo) [inline packages/frameworks/frameworks.json]: Can we mention the field name?

**Review by leo — APPROVED** [review]

File: packages/frameworks/index.d.ts [hunk 3]
```diff
@@ -13,4 +13,10 @@ export interface Framework {
     every?: FrameworkDetectionItem[];
     some?: FrameworkDetectionItem[];
   };
+  settings?: {
+    [key: string]: {
+      value?: string;
+      placeholder?: string;
```
Comment (lucleray) [inline packages/frameworks/index.d.ts]: Why not explicitly type `buildCommand`, `devCommand` and `outputDirectory`?

Comment (AndyBitz) [reply packages/frameworks/index.d.ts]: We'll add more and specifying everything is not that extensible, and in the future we'll probably make the whole frameworks.json file typescript to make sure we also know the type when importing.

Comment (leo) [reply packages/frameworks/index.d.ts]: I think we should explicitly type them – I don't see an argument against that

**Review by leo — CHANGES_REQUESTED** [review]

File: packages/frameworks/index.d.ts (hunk 3 above)
Comment (lucleray) [reply packages/frameworks/index.d.ts]: > we'll probably make the whole frameworks.json file typescript

So wouldn't it be better to be explicit? If the whole file is typescript you'll be explicit, no?

**Review by lucleray — APPROVED** [review]

**Pull request merged**